src/Config.php

```diff
@@ -86,32 +89,14 @@ public function toArray()
     }
 
     /**
-     * Copy paste from https://github.com/laminas/laminas-stdlib/commit/26fcc32a358aa08de35625736095cb2fdaced090
-     * to keep compatibility with previous version
-     *
-     * @link https://github.com/zendframework/zend-servicemanager/pull/68
+     * @psalm-param ServiceManagerConfigurationType $a
+     * @psalm-param ServiceManagerConfigurationType $b
+     * @psalm-return ServiceManagerConfigurationType
+     * @psalm-suppress MixedReturnTypeCoercion
      */
-    private function merge(array $a, array $b): array
+    private function merge(array $a, array $b)
     {
-        foreach ($b as $key => $value) {
-            if ($value instanceof MergeReplaceKeyInterface) {
-                $a[$key] = $value->getData();
-            } elseif (isset($a[$key]) || array_key_exists($key, $a)) {
-                if ($value instanceof MergeRemoveKey) {
-                    unset($a[$key]);
-                } elseif (is_int($key)) {
-                    $a[] = $value;
-                } elseif (is_array($value) && is_array($a[$key])) {
-                    $a[$key] = $this->merge($a[$key], $value);
-                } else {
-                    $a[$key] = $value;
-                }
-            } else {
-                if (! $value instanceof MergeRemoveKey) {
-                    $a[$key] = $value;
-                }
-            }
-        }
-        return $a;
+        /** @psalm-suppress MixedReturnTypeCoercion */
+        return ArrayUtils::merge($a, $b);
```

Comment:
I removed the inlined code here as `laminas-stdlib` is a dependency within `ServiceManager` to use `merge` for the `lazy_services`.